consensus/expected.go

```diff
@@ -238,9 +238,12 @@ func (c *Expected) IsHeavier(ctx context.Context, a, b types.TipSet, aSt, bSt st
 // starting state and a tipset to a new state.  It errors if the tipset was not
 // mined according to the EC rules, or if running the messages in the tipset
 // results in an error.
-func (c *Expected) RunStateTransition(ctx context.Context, ts types.TipSet, ancestors []types.TipSet, pSt state.Tree) (state.Tree, error) {
-	err := c.validateMining(ctx, pSt, ts, ancestors[0])
-	if err != nil {
+func (c *Expected) RunStateTransition(ctx context.Context, ts types.TipSet, ancestors []types.TipSet, pSt state.Tree) (st state.Tree, err error) {
+	ctx, span := trace.StartSpan(ctx, "Expected.RunStateTransition")
+	span.AddAttributes(trace.StringAttribute("tipset", ts.String()))
+	defer tracing.AddErrorEndSpan(ctx, span, &err)
+
+	if err := c.validateMining(ctx, pSt, ts, ancestors[0]); err != nil {
```

Comment:
Blocking: I'm suspicious that this `:=` means that this error is not properly being set to the named return value preventing the span from properly reporting the error.

I could be wrong, ignore if you're sure this is fine.

Reply:
I think its fine: https://play.golang.org/p/9vJGxElgkZu

Reply:
Ah this is super interesting. It looks like if you name a variable, regardless of how it's returned (by assignment, or directly in a return state), it gets correctly assigned.